src/iterable_recursive_open_struct.rb

```diff
@@ -0,0 +1,26 @@
+
+module Metalware
+  class IterableRecursiveOpenStruct < RecursiveOpenStruct
+    def each(&block)
+      convert_hash_values_to_own_class.each(&block)
+    end
+
+    def each=(*args)
+      raise IterableRecursiveOpenStructPropertyError,
+        "Cannot set property 'each', reserved to use for iteration"
+    end
+
+    private
+
+    def convert_hash_values_to_own_class
```

Comment:
Not sure what you mean - it's private so shouldn't conflict with anything if someone does choose to use this name I think? E.g.:

```ruby
[25] pry(main)> a = Metalware::IterableRecursiveOpenStruct.new
=> #<Metalware::IterableRecursiveOpenStruct>
[26] pry(main)> a.convert_hash_values_to_own_class
=> nil
[27] pry(main)> a.convert_hash_values_to_own_class = 'foo'
=> "foo"
[28] pry(main)> a.convert_hash_values_to_own_class
=> "foo"
```